In [4]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error, r2_score, mean_absolute_error

df = pd.read_csv('clean_data.csv')

X = df.drop('Price', axis=1)
y = df['Price']

# Split the data into training and testing sets (80/20 split)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=.2, random_state=42)

In [11]:
from sklearn.tree import DecisionTreeRegressor

dtree_model = DecisionTreeRegressor(random_state=42)

# Fit the model on the training data
dtree_model.fit(X_train, y_train)

# Predict on the test data
y_pred_dtree = dtree_model.predict(X_test)

# Evaluate Performance
print(f"MSE: {mean_squared_error(y_test, y_pred_dtree)}")
print(f"RMSE: {mean_squared_error(y_test, y_pred_dtree, squared=False)}")
print(f"MAE: {mean_absolute_error(y_test, y_pred_dtree)}")
print(f"R-squared: {r2_score(y_test, y_pred_dtree)}")

MSE: 136339090239.02507
RMSE: 369241.23583238246
MAE: 225592.9098053892
R-squared: 0.6548037348521415


In [12]:
# Predict on the test data
y_pred_dtree = dtree_model.predict(X_train)

# Evaluate Performance
print(f"MSE: {mean_squared_error(y_train, y_pred_dtree)}")
print(f"RMSE: {mean_squared_error(y_train, y_pred_dtree, squared=False)}")
print(f"MAE: {mean_absolute_error(y_train, y_pred_dtree)}")
print(f"R-squared: {r2_score(y_train, y_pred_dtree)}")

MSE: 394509.68466361
RMSE: 628.1000594360822
MAE: 14.363245064096565
R-squared: 0.9999989216732428


Shows the scores of each model (Testing and then Training)

Linear Regression (Same accuracy on both - not great - underfitting)

MSE: 138194061946.39297
RMSE: 371744.61925681314
MAE: 251502.86452095807
R-squared: 0.6501071412030586


MSE: 124875285862.72173
RMSE: 353376.97415468615
MAE: 245491.0076728736
R-squared: 0.6586741281786699

KNN (Better accuracy on Training but not by insane amount - slight overfitting)

MSE: 129542766532.88054
RMSE: 359920.5002953854
MAE: 228576.79461077845
R-squared: 0.6720113130748206

MSE: 81861044189.84785
RMSE: 286113.69102132786
MAE: 184551.08515018245
R-squared: 0.7762464199119372

DTREE (Almost perfect on Training, not great on Testing - Overfitting)

MSE: 136339090239.02507
RMSE: 369241.23583238246
MAE: 225592.9098053892
R-squared: 0.6548037348521415

MSE: 394509.68466361
RMSE: 628.1000594360822
MAE: 14.363245064096565
R-squared: 0.9999989216732428

XGBoost (Very good score on both but better on Testing - very slight overfitting)

MSE: 64681457988.97476
RMSE: 254325.4961441632
MAE: 160174.89092323167
R-squared: 0.8362333379006134

MSE: 17860879058.60597
RMSE: 133644.59981086393
MAE: 91766.92729776831
R-squared: 0.951180250967546

